src/controller/CommissionerDiscoveryController.h

```diff
@@ -235,6 +235,19 @@ class DLL_EXPORT AppInstallationService
      */
     virtual bool LookupTargetContentApp(uint16_t vendorId, uint16_t productId) = 0;
 
+    /**
+     * @brief
+     *   Called to check app's installation status.
+     *
+     * This will be called by the main chip thread so any blocking work should be moved to a separate thread.
+     *
+     *  @param[in]    vendorId           The vendorId in the DNS-SD advertisement of the requesting commissionee.
+     *  @param[in]    productId          The productId in the DNS-SD advertisement of the requesting commissionee.
+     *
+     */
+    virtual chip::Protocols::UserDirectedCommissioning::CommissionerDeclaration::CdError
```

Comment:
It's not intuitive to have this return a CdError. So if we are re-using that type then we need to provide explanation for how to interpret the different values like NoError.